inference/core/registries/roboflow.py

```diff
@@ -80,26 +85,38 @@ def get_model_type(model_id: str, api_key: str) -> Tuple[TaskType, ModelType]:
     )
     if cached_metadata is not None:
         return cached_metadata[0], cached_metadata[1]
-    workspace_id = get_roboflow_workspace(api_key=api_key)
-    project_task_type = get_roboflow_dataset_type(
-        api_key=api_key, workspace_id=workspace_id, dataset_id=dataset_id
-    )
     if version_id == STUB_VERSION_ID:
+        if api_key is None:
+            raise MissingApiKeyError(
+                "Stub model version provided but no API key was provided. API key is required to load stub models."
+            )
+        workspace_id = get_roboflow_workspace(api_key=api_key)
+        project_task_type = get_roboflow_dataset_type(
+            api_key=api_key, workspace_id=workspace_id, dataset_id=dataset_id
+        )
         model_type = "stub"
-    else:
-        model_type = get_roboflow_model_type(
-            api_key=api_key,
-            workspace_id=workspace_id,
+        save_model_metadata_in_cache(
             dataset_id=dataset_id,
             version_id=version_id,
             project_task_type=project_task_type,
+            model_type=model_type,
         )
+        return project_task_type, model_type
+    api_data = get_roboflow_model_data(
+        api_key=api_key,
+        model_id=model_id,
+        endpoint_type=ModelEndpointType.ORT,
+        device_id=GLOBAL_DEVICE_ID,
+    ).get("ort")
+    project_task_type = api_data.get("type", "object-detection")
+    model_type = api_data.get("modelType")
```

Comment:
as per typing `model_type` shall not be None - so either we mark type as none or raise error (like `ModelArtefactError`)